### PR TITLE
ci: add `clippy` job to check for linting issues

### PR DESCRIPTION
**Pull request by hamirmahal**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,94 +11,100 @@ on:
...
       version: ${{ steps.crate_metadata.outputs.version }}
       maintainer: ${{ steps.crate_metadata.outputs.maintainer }}
       homepage: ${{ steps.crate_metadata.outputs.homepage }}
       msrv: ${{ steps.crate_metadata.outputs.msrv }}
 
+  clippy:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+      - run: cargo clippy -- --allow clippy::result_large_err --deny warnings
```
[the remark this answers is not in the document]
Comment (hamirmahal) [reply .github/workflows/ci.yml:41]: I figured we could protect against other linting issues in the meantime.

Comment (hamirmahal) [reply .github/workflows/ci.yml:41]: `clippy::result_large_err` takes more work to resolve.

**Comment by mhmdanas** [comment]
Can you please revert the gratuitous style changes?

**Comment by sharkdp** [comment]
I'd rather not have this as a blocking CI check, to be honest.

**Comment by hamirmahal** [comment]
Okay, it's up to you.

**Pull request closed without merging**